sktime/classification/interval_based/_tsf.py

```diff
@@ -97,6 +109,66 @@ def __init__(self,
         # We need to add is-fitted state when inheriting from scikit-learn
         self._is_fitted = False
 
+    @classmethod
+    def _get_intervals(
+        cls, n_intervals, min_interval, series_length, random_state=None
+    ):
+        """
+        Generate random intervals for given parameters.
+        """
+        rng = check_random_state(random_state)
+        intervals = np.zeros((n_intervals, 2), dtype=int)
+        for j in range(n_intervals):
+            intervals[j][0] = rng.randint(series_length - min_interval)
+            length = rng.randint(series_length - intervals[j][0] - 1)
+            if length < min_interval:
+                length = min_interval
+            intervals[j][1] = intervals[j][0] + length
+        return intervals
+
+    @classmethod
```

Comment:
Same here, why do you use @classmethod here? :)

Reply:
Hey, I used **@classmethod** here and in a bunch of other places to provide a way to modify the algorithm behavior through subclassing. If you think that won't be necessary or should not happen I can refactor most of this to be standalone methods?

Reply:
Why not use staticmethod for subclassing? Subclassing may actually be relevant if you look at the RandomIntervalSpectralEnsemble, which essentially only applies different transformations. You could also easily convert these algorithms into regressors by simply changing the final decision tree to a regressor.

Reply:
I agree I could use a mix of @classmethod, @staticmethod and plain methods. Maybe something like this:
`_get_intervals` - static of plain method, depending on if someone might want to select intervals some other way (probably not),
`_get_transformed_data` - static,
`_fit_estimator` - class, as it needs to call proper `_get_transformed_data` when inheriting,
`_predict_proba_for_estimator` - same as above,
`_lsq_fit` - plain

Reply:
I still don't understand why we need a class method for inheritance to work, even if it's a plain method it can call `self._get_tranformed_data` no? Perhaps I'm missing something!

Reply:
It could be either a classmethod or a plain method, but not static or stand-alone method, and I thought it's better to use classmethod as plain methods are serialized with the whole object (if I am not mistaken) when passed to other process, whereas I would expect classmethods to only go with the class object (not so sure about it though). Also I mixed up some names in my previous comment (sorry!) What I meant was:

`_get_intervals` - static method or ~~plain~~ stand-alone function, depending on if someone might want to select intervals some other way (probably not),
`_get_transformed_data` - static,
`_fit_estimator` - class, as it needs to call proper _get_transformed_data when inheriting,
`_predict_proba_for_estimator` - same as above,
`_lsq_fit` - ~~plain~~ stand-alone function

Reply:
Not sure about serialisation, just always find it strange to see `cls.method` instead of `self.method`. I'd try to make it as simple as possible and stand-alone functions are probably the simplest construct, class method the most complex and the others in between, no? So I'd suggest to make them all stand-alone functions unless you want to also refactor the `RandomIntervalSpectralForest` or a `TimeSeriesForestRegressor` :)

Reply:
OK, then. I'll refactor them to be stand-alone functions for now. Let's not introduce unnecessary complexity. If the need for subclassing etc arises, it could be reverted. Not sure if I have the bandwidth to refactor the other classes but who knows :)

Reply:
Sure, thanks! Ping me when I should review again!

Reply:
Hey, I think it's ready.